src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/ActionBar.xaml.cs

```diff
@@ -42,4 +44,6 @@ private void ActionListViewItem_Tapped(object sender, TappedRoutedEventArgs e)
     }
 
     public void Receive(UpdateActionBarMessage message) => ViewModel.SelectedItem = message.ViewModel;
+
+    public void Receive(UpdateActionBarPage message) => ViewModel.CurrentPage = message.Page;
```

Comment:
If all we're doing in the view is routing the messages to change properties in the VM, then we should just have the ViewModel listen to the messages directly to update itself, makes it easier to see the connection vs. it being obscured by the view layer (same with the selection changed message in addition to the changed page).

If we're going to have a single ViewModel that we're updating like this too, then it means we don't have to worry about the page listening/deactivating in the View as much and can manage that more centrally from the ViewModel as well. If we need to do View specific things later, we can additionally re-register them here as well.

Reply:
Okay that makes sense. We hadn't done any ViewModel -> ViewModel messages before, but if it's totally allowed, then I'll just do it. Just wanted to be sure ☺️